src/ol/render/webgl/webglreplay.js

```diff
@@ -1011,27 +1028,180 @@ goog.inherits(ol.render.webgl.LineStringReplay, ol.render.webgl.Replay);
+    this.vertices_[numVertices++] = p1[0];
+    this.vertices_[numVertices++] = p1[1];
+    this.vertices_[numVertices++] = p2[0];
+    this.vertices_[numVertices++] = p2[1];
+    this.vertices_[numVertices++] = sign;
+    this.vertices_[numVertices++] = ol.render.webgl.LineStringInstruction.BEVEL_SECOND;
+
+    this.vertices_[numVertices++] = p0[0];
+    this.vertices_[numVertices++] = p0[1];
+    this.vertices_[numVertices++] = p1[0];
+    this.vertices_[numVertices++] = p1[1];
+    this.vertices_[numVertices++] = p2[0];
+    this.vertices_[numVertices++] = p2[1];
+    this.vertices_[numVertices++] = -sign;
+    this.vertices_[numVertices++] = ol.render.webgl.LineStringInstruction.MITER_BOTTOM;
 
-  // Shift the indices to take into account previously handled lines
-  for (i = offset, ii = end - stride; i < ii; i += stride) {
+    this.indices_[numIndices++] = n;
+    this.indices_[numIndices++] = lastIndex - 1;
+    this.indices_[numIndices++] = lastIndex;
 
-    if (i == offset) {
-      this.vertices_[numVertices++] = flatCoordinates[i] - this.origin_[0];
-      this.vertices_[numVertices++] = flatCoordinates[i + 1] - this.origin_[1];
+    this.indices_[numIndices++] = n + 2;
+    this.indices_[numIndices++] = n;
+    this.indices_[numIndices++] = lastSign * sign > 0 ? lastIndex : lastIndex - 1;
 
-      i += stride;
-    }
+    this.indices_[numIndices++] = n;
+    this.indices_[numIndices++] = n + 2;
+    this.indices_[numIndices++] = n + 1;
 
-    this.vertices_[numVertices++] = flatCoordinates[i] - this.origin_[0];
-    this.vertices_[numVertices++] = flatCoordinates[i + 1] - this.origin_[1];
+    lastIndex = n + 2;
+    lastSign = sign;
+
+    //Add miter
+    if (verticesPerPoint === 4) {
+      this.vertices_[numVertices++] = p0[0];
+      this.vertices_[numVertices++] = p0[1];
+      this.vertices_[numVertices++] = p1[0];
+      this.vertices_[numVertices++] = p1[1];
+      this.vertices_[numVertices++] = p2[0];
+      this.vertices_[numVertices++] = p2[1];
+      this.vertices_[numVertices++] = sign;
+      this.vertices_[numVertices++] = ol.render.webgl.LineStringInstruction.MITER_TOP;
+
+      this.indices_[numIndices++] = n + 1;
+      this.indices_[numIndices++] = n + 3;
+      this.indices_[numIndices++] = n;
+    }
+  }
+};
 
-    this.indices_[numIndices++] = n++;
-    this.indices_[numIndices++] = n;
+/**
+ * Check if the linestring is a boundary.
+ * @param {Array.<number>} flatCoordinates Flat coordinates.
+ * @param {number} offset Offset.
+ * @param {number} end End.
+ * @param {number} stride Stride.
+ * @return {boolean} The linestring is a boundary.
+ * @private
+ */
+ol.render.webgl.LineStringReplay.prototype.isClosed_ = function(flatCoordinates, offset, end, stride) {
+  var lastCoord = end - stride;
+  if (flatCoordinates[offset] === flatCoordinates[lastCoord] &&
+      flatCoordinates[offset + 1] === flatCoordinates[lastCoord + 1]) {
+    return true;
   }
+  return false;
 };
 
```

Comment:
This is a helper function used only by the LineString WebGL renderer. On the other hand, it is a general function working on flat geometries. Should it go to the ol.geom.flat namespace?